examples/multi-dl.rs

```diff
@@ -0,0 +1,83 @@
+use std::collections::HashMap;
+use std::time::Duration;
+
+use anyhow::Result;
+
+use curl::easy::{Easy2, Handler, WriteError};
+use curl::multi::{Easy2Handle, Multi};
+
+const URLS: &[&str] = &[
+    "https://www.microsoft.com",
+    "https://www.google.com",
+    "https://www.amazon.com",
+    "https://www.apple.com",
+];
+
+struct Collector(Vec<u8>);
+impl Handler for Collector {
+    fn write(&mut self, data: &[u8]) -> Result<usize, WriteError> {
+        self.0.extend_from_slice(data);
+        Ok(data.len())
+    }
+}
+
+fn download(multi: &mut Multi, token: usize, url: &str) -> Result<Easy2Handle<Collector>> {
+    let version = curl::Version::get();
+    let mut request = Easy2::new(Collector(Vec::new()));
+    request.url(&url)?;
+    request.useragent(&format!("curl/{}", version.version()))?;
+
+    let mut handle = multi.add2(request)?;
+    handle.set_token(token)?;
+    Ok(handle)
+}
+
+fn main() -> Result<()> {
+    let mut multi = Multi::new();
+    let mut handles: HashMap<_, _> = URLS
+        .iter()
+        .enumerate()
+        .map(|(token, url)| (token, download(&mut multi, token, url).unwrap()))
```

Comment:
Could this `.unwrap()` instead propagate the result?

Reply:
I'm not sure I understand correctly which part should unwrap?

Do you mean unwrapping in `download` such that the return value is an `Easy2Handle`?